modules/bene_core/src/Plugin/Block/SubNavigationBlock.php

```diff
@@ -0,0 +1,178 @@
+    $breadcrumb_links = [];
+    /** @var Link $breadcrumb */
+    foreach ($breadcrumbs as $breadcrumb) {
+      $breadcrumb_links[] = Link::fromTextAndUrl($breadcrumb->getText(), $breadcrumb->getUrl());
+    }
+
+    $build['breadcrumbs'] = [
+      '#theme' => 'item_list',
+      '#wrapper_attributes' => [
+        'class' => [
+          'bene-subnav-breadcrumbs',
+        ],
+      ],
+      '#items' => $breadcrumb_links,
+    ];
+
+    // Add current page title.
+    $page_title = $this->getPageTitle();
+    // Account for the page title being returned as either a string or render
+    // array.
+    if (is_array($page_title)) {
+      $build['page_title'] = $page_title;
+    }
+    else {
+      $build['page_title'] = [
+        '#type' => 'markup',
+        '#markup' => $this->getPageTitle(),
+        '#prefix' => '<div class="bene-subnav-page-title">',
+        '#suffix' => '</div>',
+      ];
+    }
+
+    $page_child_links = [];
+
+    $page_children = $this->getChildMenuLinks();
+    if (!empty($page_children)) {
+      foreach ($page_children as $page_child) {
+        $page_child_links[] = Link::createFromRoute($page_child['title'], $page_child['route_name'], $page_child['route_parameters']);
+      }
+
+      $build['page_children'] = [
+        '#theme' => 'item_list',
+        '#wrapper_attributes' => [
+          'class' => [
+            'bene-subnav-children',
+          ],
+        ],
+        '#items' => $page_child_links,
+      ];
+    }
+
+    return $build;
+  }
+
+  /**
+   * Gets an array of breadcrumb links up to (not including) the current page.
+   *
+   * @return array
+   *   Array of Link objects representing breadcrumbs.
+   */
+  private function getBreadcrumbs() {
+    /** @var \Drupal\Core\Routing\RouteMatch $current_route */
+    $current_route = \Drupal::service('current_route_match')->getCurrentRouteMatch();
+    $breadcrumbs = \Drupal::service('breadcrumb')->build($current_route)->getLinks();
+
+    return $breadcrumbs;
+  }
+
+  /**
+   * Gets the title of the current page.
+   *
+   * @return array or string
+   *   The page title as a string or render array.
+   *
+   * @see https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Controller%21TitleResolver.php/class/TitleResolver/8.2.x
+   */
+  private function getPageTitle() {
+    /** @var \Drupal\Core\Routing\RouteMatch $current_route */
+    $current_route = \Drupal::service('current_route_match')->getCurrentRouteMatch();
+    $title = \Drupal::service('title_resolver')->getTitle(\Drupal::request(), $current_route->getRouteObject());
+
+    return $title;
+  }
+
+  /**
+   * Gets an array of menu link definitions for child nodes of the current node.
+   *
+   * Based on the current node's menu link.
+   *
+   * @return array
+   *   Array of menu link definitions (each definition is in array format).
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  private function getChildMenuLinks() {
```

Comment:
This is assuming nodes, what happens if a contact form or other entity is placed in a menu?